cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -0,0 +1,85 @@
+package cloud.nio.impl.drs
+
+import com.typesafe.config.Config
+import org.apache.http.{HttpStatus, StatusLine}
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpPost}
+import org.apache.http.entity.{ContentType, StringEntity}
+import org.apache.http.impl.client.{CloseableHttpClient, HttpClientBuilder}
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.apache.http.util.EntityUtils
+import spray.json._
+
+import scala.util.{Failure, Success, Try}
+
+
+case class DrsPathResolver(config: Config) {
+  private val DrsPathToken = s"$${drsPath}"
+  private lazy val marthaUri = config.getString("martha.url")
+  private lazy val marthaRequestJsonTemplate = config.getString("martha.request.json-template")
+
+
+  def makeHttpRequestToMartha(drsPath: String, httpClient: CloseableHttpClient): CloseableHttpResponse = {
+    val postRequest = new HttpPost(marthaUri)
+    val requestJson = marthaRequestJsonTemplate.replace(DrsPathToken, drsPath)
+    postRequest.setEntity(new StringEntity(requestJson, ContentType.APPLICATION_JSON))
+    httpClient.execute(postRequest)
+  }
+
+
+  def resolveDrsThroughMartha(drsPath: String): MarthaResponse = {
```

Comment:
Could you add a scala doc on this to say that this method makes a synchronous HTTP request to Martha just so that it's clear from the caller side ?